tests/executables_src/testLMapMathPluginPushPars.cc

```diff
@@ -0,0 +1,125 @@
+
+#include "Device.h"
+#include "LogicalNameMappingBackend.h"
+
+using namespace ChimeraTK;
+
+BOOST_AUTO_TEST_SUITE(LMapMathPluginTestSuite)
+
+/********************************************************************************************************************/
+
+struct DummyForCleanupCheck : public LogicalNameMappingBackend {
+  using LogicalNameMappingBackend::LogicalNameMappingBackend;
+
+  static boost::shared_ptr<DeviceBackend> createInstance(std::string, std::map<std::string, std::string>) {
+    return boost::make_shared<DummyForCleanupCheck>("mathPluginWithPushPars.xlmap");
+  }
+  ~DummyForCleanupCheck() override {
+    std::cout << "~DummyForCleanupCheck()" << std::endl;
+    cleanupCalled = true;
+  }
+
+  struct BackendRegisterer {
+    BackendRegisterer() {
+      ChimeraTK::BackendFactory::getInstance().registerBackendType(
+          "DummyForCleanupCheck", &DummyForCleanupCheck::createInstance, {"map"});
+    }
+  };
+  static std::atomic_bool cleanupCalled;
+};
+std::atomic_bool DummyForCleanupCheck::cleanupCalled;
+static DummyForCleanupCheck::BackendRegisterer gDFCCRegisterer;
+
+BOOST_AUTO_TEST_CASE(testPushPars) {
+  setDMapFilePath("mathPluginWithPushPars.dmap");
+  {
+    // we enforce that initial values from variable definition in xlmap are never used in MathPugin
+    // - therefore, in test here, we must issue write for all push-parameters
+    // - then, version number check in MathPlugin will notice valid data has been provided after open
+    // - for poll-type accessors, version number check also succeeds since they get recent version numbers anyway
+    // Test needs to call activateAsyncRead but it must be irrelevant whether before or after writes
+
+    ChimeraTK::Device targetDevice;
+    targetDevice.open("HOLD");
+
+    auto accTarget = targetDevice.getScalarRegisterAccessor<uint32_t>("HOLD0/WORD_G");
+    auto pollPar = targetDevice.getScalarRegisterAccessor<uint32_t>("HOLD0/POLLPAR");
+    pollPar = 1;
+    pollPar.write();
+
+    ChimeraTK::Device device;
+    device.open("EOD");
+    device.activateAsyncRead();
+    auto pushPar = device.getScalarRegisterAccessor<uint32_t>("DET/EXPOSURE");
+
+    pushPar = 2;
+    pushPar.write();
+
+    auto accMathWrite = device.getScalarRegisterAccessor<double>("DET/GAIN");
+    // we don't have main value (x in formula) yet, since it wasn't yet written.
+    // therefore, we expect to have no value yet for formula output (0 is default from dummy construction)
+    accTarget.readLatest();
+    BOOST_CHECK_EQUAL(int(accTarget), 0);
+
+    // write to main value and check result
+    accMathWrite = 3;
+    accMathWrite.write();
+    accTarget.readLatest();
+    BOOST_CHECK_EQUAL(int(accTarget), 100 * (int)pollPar + 10 * (int)pushPar + (int)accMathWrite);
+
+    // write to push-parameter and check result
+    // note, it's a new feature that result is completely written when write() returns.
+    pushPar = 4;
+    pushPar.write();
+    accTarget.readLatest();
```

Comment:
actually, readLatest() is even conceptually wrong here. It means "take whatever has arrived in this thread until now, whatever the other threads have done". What you would want if it was an push-type accessor and you wanted to check that it has been pushed is an explicit ``BOOST_CHECK(accTarget.readLatest());``. A simple read() is enough here, since it cannot block.

Reply:
changed it to read()